### PR TITLE
set identity of the user who started a chain (typescript)

### PR DESCRIPTION
**Pull request by char26**

### CODE REVIEW

File: javascript/implementation/LogBackedStore.ts [hunk 1]
```diff
@@ -223,12 +223,24 @@ export class LogBackedStore implements Store {
...
             await this.unlockFile();
         unlockingFunction();
-        return {
+        const chain = {
             medallion,
             chainStart,
             actorId: actorId || 0,
             claimTime,
         };
+        this.claimedChains.push(chain);
```
Comment (char26) [inline javascript/implementation/LogBackedStore.ts:232]: The chain was previously not pushed to the list of claimedChains in the logbacked store. Was that intentional?

**Review by darinmcgill — CHANGES_REQUESTED** [review]
Okay I feel like this is most of the way there, but since the identity of a chain is determined by the comment in the first commit then the identity data should be set for every chainstart as part of the add bundle operation, rather than be exposed as an additional set function call on the store.  We'll want to be able to get the identity of chains that we didn't start as well as those that we have started.

File: javascript/unit-tests/Database.test.ts [hunk 2]
```diff
@@ -15,24 +15,24 @@ it('test commit', async () => {
         ensure(allChains.length == 1);
         ensure(allChains[0][0] == commitInfo.medallion);
         ensure(allChains[0][1] == commitInfo.chainStart);
-        return "okay!";
```
Comment (char26) [inline javascript/unit-tests/Database.test.ts:18]: Unrelated to this PR, but this test was not running for MemoryStore.

File: javascript/implementation/LogBackedStore.ts (hunk 1 above)
Comment (darinmcgill) [reply javascript/implementation/LogBackedStore.ts:232]: No looks like it was an oversight.

**Review by darinmcgill — APPROVED** [review]
It would be nice to have the database only claim a chain lazily (when a user first tries to make a change) rather than requiring that a chain be claimed just when you open a database.  That could either be done in this PR or in a follow up.

**Pull request merged**